ops/pebble.py

```diff
@@ -181,6 +182,19 @@
     total=False,
 )
 
+LocalIdentityDict = typing.TypedDict('LocalIdentityDict', {'user-id': int})
+BasicIdentityDict = typing.TypedDict('BasicIdentityDict', {'password': str})
```

Comment:
Is this missing the `username` field?

https://docs.google.com/document/d/1nASgUt-piV94i1cpFsbEPRk_xFbZt8AskgL97gh1Dgg/edit?tab=t.0

Reply:
Ah I think I see it now, the `username` is implied from the identities **dictionary key**.
Right?

This is a bit... unconventional.
I understand that Golang implementation doing that way.
A YAML config file is likewise OK.
That use in the Pebble HTTP API is a bit questionable...
And having this in the Python API gives me the jibbies.

It's a bit of a slippery slope.
@benhoyt WDYT?

Reply:
Yes, I'd originally proposed having the `username` field explicit, but we decided in spec review to not have it for now, and just use the dict key. We would potentially add it as a field later (at the Pebble level) if users wanted the basic auth username to differ from the identity name, but for now they have to be the same.

So I'd like to stick with the Pebble semantics here in Python too.

Reply:
As discussed in the daily, this has been resolved.